### DISCUSSION TITLE
Burner Event marker adding extra changes

### DISCUSSION THREAD

**Discussion by brendon16** [artisan-roaster-scope/artisan#1464] [General]
Hello!

I'm having an issue where every time I make a burner change a few seconds later the software will plot a value exactly 1 unit below.
For example, if I make a burner change to 80%, 5 seconds later it will plot 79%. It does it every single time.
I do make quite a few burner changes per roast so it's making the end profile look overly cluttered and messy, not good for my OCD haha!

![Event marker issue](https://github.com/artisan-roaster-scope/artisan/assets/58932279/cb04c56c-96f6-491b-b55b-9a8adac7a84a)

**Comment by poundy** [comment]
it's your alarms most likely. Settings needed to review

**Reply by brendon16** [reply]
There's no alarms set at all

**Comment by MAKOMO** [comment]
Without information on your setup (roasting machine, Artisan machine setup, Artisan settings) it is difficult to help.

**Reply by brendon16** [reply]
I tried to upload the settings file but it wouldn’t let me.

here’s a Google drive link:
https://drive.google.com/file/d/1EvNv66UVc-We-zLdFwgdYZ1zs3hR-L_k/view?usp=drivesdk

Here’s the profile for reference:
https://drive.google.com/file/d/18DI9AsoBoEQF3pTmXAi5XD7CeGbPqacV/view?usp=drivesdk

**Reply by mrpenner** [reply]
Did you rename the settings file to end in `.txt` when you tried to upload? I think that's needed for GitHub to accept it.

**Reply by brendon16** [reply]
No I didn't try that, I don't use GitHub often enough to know that was a thing. Is the google drive links not working?

**Comment by Terracotta-6** [comment]
[artisan-settings-brendon.aset.txt](https://github.com/artisan-roaster-scope/artisan/files/14396141/artisan-settings-brendon.aset.txt)
[RM - PNG - Lufa B #7 20kg_24-02-24_1232.alog,txt](https://github.com/artisan-roaster-scope/artisan/files/14396142/RM.-.PNG.-.Lufa.B.7.20kg_24-02-24_1232.alog.txt)

**Comment by Terracotta-6** [comment]
![ArtisanScreenshot](https://github.com/artisan-roaster-scope/artisan/assets/149906183/7d1000b7-2cfe-4be4-87d2-4e122b7f5411)

**Reply by Terracotta-6** [reply]
No event labeling problem was found.

**Comment by MAKOMO** [comment]
Brandon, you did not give us any information on your machine which seems not to be one of those supported by Artisan. You have Event Quantifiers defined which observe the data sampled (every 2s in your case) and on detecting a change that goes beyond the current linspace (defined by the quantifiers value range and step size) it generates a custom event with the new linspace item. In case of your burner quantifier, it generates a linspace from 0-100 with an item size of 1. 

If you change your burner to 80% (I assume using the Artisan slider, right?) Artisan sets the register 9888 of slave 4 to 80*100=8000. Latest 2s later Artisan samples the same register 8716 and takes the received value divided by 10 as reading for the first curve of your second extra device labeled "BURNER(%)". This curve is observed by your 4th event quantifier "Burner" and depending on the reading and the current event value (80 at this moment in time) it generates a new event. Obviously your Burner reading is lower than 79.5 thus that register 8716 returns a reading below 795.

If you take a look at the data table for the profile you linked, you see that at reading 139 ( at 3m58 into the roast) you set the burner slider to 90 (or you use a corresponding event button for this), resulting in the next reading of "BURNER(%)" to fall from 97.4 to 88.3. Based on this the event quantifier would generate an "88% burner" event. Somehow your profile and your settings do not correspond here.

<img width="1024" alt="Screenshot 2024-02-25 at 14 49 41" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/0e8deccc-31fb-4aa6-a338-e1d7e56a60dd">

Take a read at [Event Quantifiers](https://artisan-roasterscope.blogspot.com/2014/04/event-quantifiers.html).

**Reply by MAKOMO** [reply]
Your "Damper" Event Quantifier will unlikely work as intended. For one its range is different to the corresponding slider its applied to and the observed channel applies a factor to the received data.

**Reply by brendon16** [reply]
Thanks for the info, the machine is a 30kg OTesla but I'm using the software settings file the manufacturer sent me for some reason when I select it from your list of Devices I can't get it to control anything. Not sure if they've updated or changed the interface board but I couldn't get it to work.

In regards to the event #139 - #140 I believe that was me pressing the button [-10 Burner] and not the random changes I was saying in my initial post. In fact now I'm looking at this profile I may have sent you the wrong one. I actually manually edited this one and removed the extra changes to use it as a cleaner background profile. I'm sorry about that, I attached a problematic one below! 

But it sounds like you're saying the software is rounding the register value up or down to the closest whole number? Is that correct, if so is there an offset or something we could add to it or a way we can turn that rounding off, or is it moreso worthwhile to try and just get the correct values in the first place? 

[Check out this profile. ](https://drive.google.com/file/d/1yaUwDpEQVAGhTgDPAaBQSSrMDjwliazl/view?usp=drive_link)

Sorry again, I gave you the wrong profile I think!

PS; The damper is a burner Overtemp thats set on the control panel, I plan on removing that slider all together anyway.

**Comment by MAKOMO** [comment]
Your machine is not supported by Artisan. Either get your machine vendor to fix this setup or alternatively work together with us to get this machine supported. This will not be cost free.

Otherwise just deactivate your quantifiers.

**Reply by brendon16** [reply]
What would the cost be to get this up and running?

**Reply by MAKOMO** [reply]
The price for a machine integration is the same for every vendor. The company can contact us via our contact form at https://artisan-roasterscope.blogspot.com/p/contact-me.html and request a quote.
